core/signals.py

```diff
@@ -18,6 +20,13 @@ def bypass_antivirus_scan_if_needed(sender, instance, **kwargs):
         instance.is_infected = False
 
 
+@receiver(pre_save, sender=Document)
+def set_mimetype_on_create(sender, instance, **kwargs):
+    if not instance.pk and not instance.mimetype and instance.file:
+        validator = MagicMimeValidator()
+        instance.mimetype = validator(instance.file)
```

Comment:
La vérification étant déjà effectuée via `validators=[validate_upload_file, ...]` dans la définition du modèle, je pense qu'on peut juste récupérer le mimetype `magic.from_buffer` : 
```suggestion
        instance.mimetype = magic.from_buffer(file.read(2048), mime=True)
```